informatics-isi-edu/chaise
# Add edit link to record page (#402)

**Changes**
- The link redirects user to the recordedit (data entry) app.
- The redirect url is determined via `"../recordedit"` + `window.location.hash`.
- The Edit link only appears if:
 1. The catalog ACL for `content_write_user` has an `*` or matches one of the user's session attribute IDs; AND
 2. Either the table or schema doesn't have an applicable `ignore` annotation; AND
 3. `chaiseConfig.editRecord` is `true`

**To Test**

1. Visit a Record page in my user dir, such as [this record from Yang Chai](https://dev.isrd.isi.edu/~jessie/chaise/record/#1/legacy:dataset/id=5776). Next to the Permalink link, click the "Edit Record" link. You should be redirected to the recordedit (data entry) app where the input fields are pre-filled with this record's values.
2. Visit a Record page that has an `ignore` annotation on its table, like [this dummy record in the `testtoo` table](https://dev.isrd.isi.edu/~jessie/chaise/record/#4/rbk:testtoo/id=1). Next to the Permlink link, there should **not** be any "Edit Record" link.

The code can be pulled down from branch `record-edit-btn`. 

~~Normally I wouldn't have worked on a separate branch for this relatively small enhancement, but I thought it would take more work than was actually necessary when I started.~~ Well now that we have the ACLs, this is no longer the case!
